### ISSUE TITLE
Getting error: cannot unmarshal array into Go struct field Message.messages.key of type string

### ISSUE THREAD

**Issue by ferras777** [mostafa/xk6-kafka#143]
Hi, @mostafa 
I get an error, when I try to serialize a key with byte_array_serializer. I'm using default script from the examples, just changed the key and value serialization. Maybe I did something incorrectly?
Error:
```shell
ERRO[0001] json: cannot unmarshal array into Go struct field Message.messages.key of type string

```
<details>
<summary style="font-size:14px">Script</summary>
<p>

```javascript
import { check } from "k6";
import {
    Writer,
    Reader,
    Connection,
    STRING_SERIALIZER,
    STRING_DESERIALIZER,
    BYTE_ARRAY_SERIALIZER,
    BYTE_ARRAY_DESERIALIZER,
} from "k6/x/kafka"; // import kafka extension

const brokers = ["localhost:9092"];
const topic = "xk6_kafka_byte_array_topic";

const writer = new Writer({
    brokers: brokers,
    topic: topic,
    autoCreateTopic: true,
});
const reader = new Reader({
    brokers: brokers,
    topic: topic,
});
const connection = new Connection({
    address: brokers[0],
});

if (__VU == 0) {
    connection.createTopic({ topic: topic });
}

var config = {
    producer: {
        keySerializer: BYTE_ARRAY_SERIALIZER,
        valueSerializer: STRING_SERIALIZER,
    },
    consumer: {
        keyDeserializer: BYTE_ARRAY_DESERIALIZER,
        valueDeserializer: STRING_DESERIALIZER,
    },
};

const payload = "byte array payload";

export default function () {
    console.log(JSON.stringify(config))
    for (let index = 0; index < 100; index++) {
        let messages = [
            {
                key: Array.from(payload, (x) => x.charCodeAt(0)),
                value: "test-id-abc-" + index,
            },
            {
                key: Array.from(payload, (x) => x.charCodeAt(0)),
                value: "test-id-def-" + index,
            },
        ];

        writer.produce({
            messages: messages,
            config: config,
        });
    }

    // Read 10 messages only
    let messages = reader.consume({
        limit: 10,
        config: config,
    });
    check(messages, {
        "10 messages returned": (msgs) => msgs.length == 10,
        "key starts with 'test-id-' string": (msgs) => msgs[0].value.startsWith("test-id-"),
        "payload is correct": (msgs) => String.fromCharCode(...msgs[0].key) === payload,
    });
}

export function teardown(data) {
    if (__VU == 0) {
        // Delete the topic
        connection.deleteTopic(topic);
    }
    writer.close();
    reader.close();
    connection.close();
}
```

</p></details>

**Comment by mostafa** [comment]
@ferras777 
I am investigating it.

**Comment by mostafa** [comment]
@ferras777 
I found the bug and [fixed](https://github.com/mostafa/xk6-kafka/commit/423aacb51efaa96534104e4febcfaedaca248308) it. I'll release a patch version soon.

**Comment by ferras777** [comment]
Thank you!

**Comment by mostafa** [comment]
Thanks for noticing and reporting it. 🙏

**Comment by mostafa** [comment]
Released [v0.13.1](https://github.com/mostafa/xk6-kafka/releases/tag/v0.13.1).
